### ISSUE TITLE
Files in node_modules are parsed on build

### ISSUE THREAD

**Issue by glecetre** [facebook/docusaurus#2213]
## 🐛 Bug Report

When running the command `npm run build`, files in the `node_modules` folder are checked by the engine.

This creates a lot of unnecessary overhead and logs many errors in the console like so:

> [WARN] unresolved links in file 'website/node_modules/mixin-deep/node_modules/is-extendable/README.md' > [ '.verb.md' ]
> [WARN] unresolved links in file 'website/node_modules/mixin-deep/README.md' > [ '.verb.md' ]
> [WARN] unresolved links in file 'website/node_modules/nanomatch/README.md' > [ '.github/contributing.md', '.verb.md' ]

There also are some errors logged from the fixture files for the tests:

> Header field "unlisted" in /home/jeahel/dev/the-hive/documentation/website/node_modules/docusaurus/lib/core/\_\_tests\_\_/\_\_fixtures\_\_/blog-post-with-draft.md is not supported.
> Header field "authorTwitter" in /home/jeahel/dev/the-hive/documentation/website/node_modules/docusaurus/lib/server/\_\_tests\_\_/\_\_fixtures\_\_/2018-08-17-docusaurus.md is not supported.

And some errors with BABEL, still in the node_modules folder:
> [BABEL] Note: The code generator has deoptimised the styling of /home/jeahel/dev/the-hive/documentation/website/node_modules/docusaurus/lib/core/metadata.js as it exceeds the max of 500KB.

### Have you read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#reporting-new-issues)?

I have.

## To Reproduce

1. Create a new Docusaurus project with `docusaurus-init`
1. Build the website with `cd website && npm run build`
1. Look at the console output

## Expected behavior

I expect the engine not to go through the `node_modules/` folder not to waste time, and no log errors about files I am not supposed/required to know about (errors about the fixtures and BABEL).

## Actual Behavior

The opposite: all files from the dependencies in `node_modules/` are parsed, and errors about unknown files are printed out.

## Reproducible Demo

No need to clone a repo, juste create a new Docusaurus project with `npx docusaurus-init`.

**Comment by glecetre** [comment]
I realized this is due to the .gitignore file so I updated the bug description.

**Comment by strtw** [comment]
Hi @glecetre, are you still seeing this issue? This is what I'm seeing: 

<img width="548" alt="Screen Shot 2020-02-10 at 10 16 01 AM" src="https://user-images.githubusercontent.com/11788126/74177564-d5d49580-4bee-11ea-8b89-af924f62e8d6.png">

**Comment by glecetre** [comment]
I'm sorry I did not come back sooner but I understood the problem.
In my config file, I had changed the `customDocsPath` setting to avoid having a `/docs` folder, so I used the value `./`. So of course there was the `website` folder, containing node_modules, and it was parsed by the engine.

Maybe you would want to change that behaviour to prevent parsing the website folder, but my bug in itself can be closed.

Thanks
